src/urllib3/contrib/wasi/wasi.py

```diff
@@ -0,0 +1,159 @@
+from typing import Optional
+from .request import WasiRequest
+from .response import BytesResponseBody, ResponseBody, WasiResponse
+import importlib
+from . import errors
+import typing
+
+C_UINT64_MAX = 18_446_744_073_709_551_615
+
+def preload(world_name: str) -> None:
+    importlib.import_module(".imports.outgoing_handler", world_name)
+    importlib.import_module(".imports.types", world_name)
+    importlib.import_module(".imports.streams", world_name)
+    importlib.import_module(".types", world_name)
+
+def send_request(request: WasiRequest, world_name: str) -> WasiResponse:
```

Comment:
This being untyped is a bit annoying, but I don't think there is a good workaround currently.

Hopefully componentize-py eventually allows generating the bindings in a static location so we can have type hints here